src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -426,20 +427,19 @@ private BoundExpression CreateTupleLiteralConversion(SyntaxNode syntax, BoundTup
             NamedTypeSymbol targetType = (NamedTypeSymbol)destinationWithoutNullable;
             if (targetType.IsTupleType)
             {
-                var destTupleType = (TupleTypeSymbol)targetType;
-
-                TupleTypeSymbol.ReportNamesMismatchesIfAny(targetType, sourceTuple, diagnostics);
+                NamedTypeSymbol.ReportTupleNamesMismatchesIfAny(targetType, sourceTuple, diagnostics);
 
                 // do not lose the original element names and locations in the literal if different from names in the target
                 //
                 // the tuple has changed the type of elements due to target-typing, 
                 // but element names has not changed and locations of their declarations 
                 // should not be confused with element locations on the target type.
-                var sourceType = sourceTuple.Type as TupleTypeSymbol;
 
-                if ((object)sourceType != null)
+                var sourceType = sourceTuple.Type as NamedTypeSymbol;
+
+                if (sourceType?.IsTupleType == true)
```

Comment:
>if (sourceType?.IsTupleType == true [](start = 16, length = 35)

Suggestion:
`if (sourceTuple.Type is NamedTypeSymbol { IsTupleType: true } sourceType)`

Reply:
You could have eliminated the declaration on the previous line too by adding a designator to the pattern match.

---
In reply to: [355070597](https://github.com/dotnet/roslyn/pull/39370#discussion_r355070597) [](ancestors = 355070597)